src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/commands/mobile/subscribe_vehicle_data_request.cc

```diff
@@ -201,20 +201,39 @@ bool SubscribeVehicleDataRequest::CheckSubscriptionStatus(
 }
 
 bool SubscribeVehicleDataRequest::SubscribePendingVehicleData(
-    ApplicationSharedPtr app, const smart_objects::SmartObject& msg_params) {
+    ApplicationSharedPtr app, smart_objects::SmartObject& msg_params) {
   LOG4CXX_DEBUG(logger_, "Subscribing to all pending VehicleData");
 
-  for (const auto& vi_name : vi_waiting_for_subscribe_) {
+  std::set<hmi_apis::Common_VehicleDataResultCode::eType> skiped_result_codes(
+      {hmi_apis::Common_VehicleDataResultCode::VDRC_TRUNCATED_DATA,
+       hmi_apis::Common_VehicleDataResultCode::VDRC_DISALLOWED,
+       hmi_apis::Common_VehicleDataResultCode::VDRC_USER_DISALLOWED,
+       hmi_apis::Common_VehicleDataResultCode::VDRC_INVALID_ID,
+       hmi_apis::Common_VehicleDataResultCode::VDRC_DATA_NOT_AVAILABLE,
+       hmi_apis::Common_VehicleDataResultCode::VDRC_DATA_NOT_SUBSCRIBED,
+       hmi_apis::Common_VehicleDataResultCode::VDRC_IGNORED});
+
+  for (auto vi_name = vi_waiting_for_subscribe_.begin();
+       vi_name != vi_waiting_for_subscribe_.end();) {
     const bool is_subscription_successful = CheckSubscriptionStatus(
```

Comment:
@ychernysheva according to naming, `CheckSubscriptionStatus` function will get the res_code of specific vehicle_info. And if it should be skipped, you will do the same afterward. How about moving the skipping check of vehicle info to this function?

Reply:
@mkorniichuk I think, that in this case is easier to check for subscription repeatedly with minor changes to base logic. If I move this part to CheckSubscriptionStatus, I'll still need to add some check in SubscribePendingVehicleData to separate skiped result codes from invalid result codes, in other case it can affect logic of generation generic error message, and this issue became more complicated.

Reply:
@ychernysheva now result code will be checked inside CheckSubscriptionStatus and then further once more. This leads to code duplications and complications in further bugfix, from my perspective. I still would suggest move this logic to `CheckSubscriptionStatus` and make some enum as a result, or remove `CheckSubscriptionStatus` at all. But it is up to you.